File: backend/src/main/kotlin/com/example/backend/controllers/AppSettingController.kt
package com.example.backend.controllers

import com.example.backend.models.AppSetting
import com.example.backend.models.BonCaisse
import com.example.backend.services.AppSettingService
import com.example.backend.services.BonCaisseService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/admin/setting")
class AppSettingController(
  private val appSettingService: AppSettingService
) {

  @GetMapping("/{key}")
  fun getParam(@PathVariable key: String): ResponseEntity<Map<String,Any?>>{
    var data = appSettingService.getParam(key)
    return ResponseEntity.ok(mapOf("key" to data))
  }

  @PostMapping("/{key}")
  fun setParam(@PathVariable key: String, @RequestBody value: String): ResponseEntity<AppSetting>{
    return ResponseEntity.ok(appSettingService.updateParam(key, value))
  }

  @GetMapping
  fun getAll(): ResponseEntity<List<AppSetting>> {
    return ResponseEntity.ok(appSettingService.getAll())
  }

}

File: backend_py/pharmaPython/app/api/v1/app_setting_controller.py
from fastapi import APIRouter, Depends, HTTPException, Body
from sqlalchemy.orm import Session
from typing import Any, Dict, List
from app.api.deps import get_db
from app.models.app_setting import AppSettingSchema
from app.services.app_setting_service import AppSettingService
from app.utility.jwt_authentication import jwt_authentication

router = APIRouter(
  prefix="/admin/setting",
  tags=["AppSetting"],
  dependencies=[Depends(jwt_authentication)],
)


@router.get("/{key}", response_model=Dict[str, Any])
def get_param(key: str, db: Session = Depends(get_db)):
  service = AppSettingService(db)
  data = service.get_param(key)
  if data is None:
    raise HTTPException(status_code=404, detail=f"Paramètre '{key}' introuvable")
  return {"key": data}


@router.post("/{key}", response_model=AppSettingSchema)
def set_param(key: str, value: str = Body(...), db: Session = Depends(get_db)):
  service = AppSettingService(db)
  return service.update_param(key, value)


@router.get("")
def get_all(db: Session = Depends(get_db)):
  service = AppSettingService(db)
  return service.get_all()
